Add a client API version bump checklist (#3135)
wireapp/wire-server: docs/src/developer/developer/api-versioning.md

different endpoints (and have different names) on the same path, and with
non-overlapping version ranges.

### Version bump checklist

When making the client API version bump, i.e., when finalising a version, there
are several steps to make apart from deciding what endpoint changes are part of
the version:

 - In `wire-api` extend the `Version` type with a new version by appending the
   new version to the end, e.g., by adding `V4`. Make sure to update its
   `ToSchema` instance,
 - In the same `Version` module update the `developmentVersions` value to list
   only the new version,

>  - Should it be desired, update the `backendApiVersion` value in Stern, which is

Reviewer: ```suggestion
 - Consider updating the `backendApiVersion` value in Stern, which is
```

it's sometimes desired and sometimes required.  :)
